### PR TITLE
correct exporter user-agent example

### PR DESCRIPTION
**Pull request by robbkidd**
## Changes

Removes spaces from the example exporter User-Agent string to conform to header field value component definition.

While this section about a User-Agent references [RFC7231](https://datatracker.ietf.org/doc/html/rfc7231#section-5.5.3) for format, the example provided includes spaces in the `product` identifier. The format for `product` is as a `token`, defined in [RFC7230](https://datatracker.ietf.org/doc/html/rfc7230#section-3.2.6), which does not include space in the set of valid characters. Including spaces here will throw off most existing user-agent parsing.

Related issues: example added in #2684

### CODE REVIEW

**Comment by arminru** [comment]
Thank you!

Please also add an entry in `CHANGELOG.md` so SDK and exporter implementors are made aware of the fix, even if it's just in the example.

**Review by arminru — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,9 @@ release.
 
 ## Unreleased
 
+- Remove spaces from example exporter User-Agent header to conform to RFC7231 & RFC7230.
```
Comment (carlosalberto) [inline CHANGELOG.md:10]: I don't think this needs/deserves a CHANGELOG entry (also, this section should be empty, as only the subsections are used 😄 )

**Comment by carlosalberto** [comment]
cc @codeboten

**Review by codeboten — APPROVED** [review]

**Review by MrAlias — APPROVED** [review]

**Comment by robbkidd** [comment]
I am unclear on what change I should make: remove the changelog entry or move it to a different section?

If a different section, which section? **OpenTelemetry Protocol** because this is about OTLP exporters?

**Comment by carlosalberto** [comment]
@robbkidd Oh, I meant removing the entry from the Changelog basically. On a second though, we want to be explicit to SIGs about updating their user agents, so leets keep keep the entry.

**Review by carlosalberto — APPROVED** [review]

**Pull request merged**